orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/StageDefinitionBuilders.kt

```diff
@@ -32,7 +32,7 @@ import com.netflix.spinnaker.orca.pipeline.model.SyntheticStageOwner.STAGE_BEFOR
  */
 fun StageDefinitionBuilder.buildTasks(stage: Stage<*>) {
   val taskGraph =
-    if (this is BranchingStageDefinitionBuilder && stage.getParentStageId() == null) {
+    if (this is BranchingStageDefinitionBuilder && (stage.getParentStageId() == null || stage.parent().getType() != stage.getType())) {
```

Comment:
Canaries _do_ have parent stages but are also parallel so I had to modify this condition. If the parent type is the same then we're in a situation like bake where the stage does double duty as a parallel fork _and_ the branches that fork creates.